src/doom/p_inter.c

```diff
@@ -207,10 +210,7 @@ P_GiveWeapon
     {
 	// give one clip with a dropped weapon,
 	// two clips with a found weapon
-	if (dropped)
-	    gaveammo = P_GiveAmmo (player, weaponinfo[weapon].ammo, 1);
-	else
-	    gaveammo = P_GiveAmmo (player, weaponinfo[weapon].ammo, 2);
+		gaveammo = P_GiveAmmo (player, weaponinfo[weapon].ammo, 2, dropped);
```

Comment:
You could simply disable the `if (dropped)` code path by appending `&& false` to the condition.

Reply:
The block is restored in the last commit, but I'll keep this in mind if you decide to have 'dropped' back on P_GiveAmmo.